Backend-API/src/main/java/com/avangers/backendapi/controllers/PetController.java

```diff
@@ -23,10 +25,24 @@ public class PetController {
     private final UserService userService;
 
     @GetMapping("/my-pets")
-    public List<PetDTO> getMyPets(@AuthenticationPrincipal UserDetails userDetails) {
+
+    public List<PetDTO> getMyPets(Principal principal) {
         // gets UserDTO using the username (email)
-        FindUserResponseDTO user = userService.findUserByEmail(userDetails.getUsername());
+        FindUserResponseDTO user = userService.findUserByEmail(principal.getName());
         // Now use the user's ID to get the pets
         return petService.getPetsByOwnerId(Long.valueOf(user.getId()));
     }
+
+    @PostMapping("/add")
```

Comment:
We should not indicate the intent with the path, we should rather use the HTTP verbs instead (which you already do).

Reply:
i am not sure, what do you mean :( What are the HTTP verbs?

Reply:
maybe ?

```suggestion
    @PostMapping // this endpoint to add a new pet
    public ResponseEntity<?> addPet(Principal principal, @Valid @RequestBody AddPetRequestDTO addPetRequestDTO) {
        try {
            AddPetResponseDTO responseAddPet = petService.addPet(addPetRequestDTO, principal.getName());
            return new ResponseEntity<>(responseAddPet, HttpStatus.CREATED);
        } catch (IllegalArgumentException e) {
            HashMap<String, String> error = new HashMap<>();
            error.put("error", e.getMessage());
            return new ResponseEntity<>(error, HttpStatus.BAD_REQUEST);
        }
    }
```

Reply:
I meant you do not need to add the "add", you should use the post rather as a method.